fastlane/lanes/build.rb

```diff
@@ -0,0 +1,427 @@
+# frozen_string_literal: true
+
+# Lanes related to Building and Testing the code
+#
+platform :ios do
+  # Builds the WordPress app for Testing
+  #
+  # @called_by CI
```

Comment:
The intent was mostly to identify lanes that are called by CI vs lanes that we typically run from our machines, in order to help have a better mental model of what calls what (eg code freeze is called locally, triggers CI, then CI is the one typically calling the build lane).

It was mostly helping me make sense of which lanes typically serves which purpose and which lane was calling which other lane while I was documenting them. Which helped me notice the uselessness of `build_and_upload_itc` above for example. And then I figured since I was trying to make sense of all that call tree on my notepad… I might as well document it with those `@called_by other_lane` annotations to help users figure out which were the "root lanes" vs which were secondary ones. Then `@called_by CI` came naturally when documenting when the caller was not another lane but a `.sh` called by `.buildkite/pipeline.yml`

Ultimately I agree that the fact that a lane is called by CI via pipeline.yml does not mean we can't call it manually too if we want (that was kind of implied from it being a public lane and not a `private_lane` or `def`). But I also thought that leaving this in the doc comment could signal that if we ever change the name of the lane or its expected options we should make sure to update the call made by CI in the `.buildkite/*/*.sh` too 🤷‍♂️

TL;DR: I would not mind if you prefer that I remove those, but figured they wouldn't hurt and could be a useful indicator, even if that isn't intended to mean that we can't call them manually too.

Reply:
> figured they wouldn't hurt and could be a useful indicator, even if that isn't intended to mean that we can't call them manually too.

Okay, fair. I worry they might get out of date, like all comments tend to do. Then again, ideally, we'll call _more_ lanes from CI, not less. Maybe we'll get to a point where it's only useful to highlight a few outlier lanes mean for local dev, and this annotation will become redundant. Till then, okay to keep.